packages/lesswrong/lib/collections/tagFlags/collection.ts

```diff
@@ -0,0 +1,76 @@
+import { createCollection, Utils } from '../../vulcan-lib';
+import { addUniversalFields, getDefaultResolvers, getDefaultMutations, schemaDefaultValue } from '../../collectionUtils'
+import { SchemaType } from '../../utils/schemaUtils'
+import { makeEditable } from '../../editor/make_editable';
+import './fragments'
+
+
+const schema: SchemaType<DbTagFlag> = {
+  createdAt: {
+    optional: true,
+    type: Date,
+    canRead: ['guests'],
+    onInsert: () => new Date(),
+  },
+  name: {
+    type: String,
+    canRead: ['guests'],
+    canUpdate: ['members', 'admins', 'sunshineRegiment'],
+    canCreate: ['members', 'admins', 'sunshineRegiment'],
+    order: 1
+  },
+  deleted: {
+    optional: true,
+    type: Boolean,
+    canRead: ['guests'],
+    canUpdate: ['admins', 'sunshineRegiment'],
+    canCreate: ['admins', 'sunshineRegiment'], 
+    order: 2,
+    ...schemaDefaultValue(false),
+  },
+  slug: {
+    type: String,
+    optional: true,
+    viewableBy: ['guests'],
+    onInsert: (tagFlag) => {
+      return Utils.getUnusedSlugByCollectionName("TagFlags", Utils.slugify(tagFlag.name))
+    },
+    onEdit: (modifier, tagFlag) => {
+      if (modifier.$set.name) {
+        return Utils.getUnusedSlugByCollectionName("TagFlags", Utils.slugify(modifier.$set.name), false, tagFlag._id)
+      }
+    }
+  },
+  order: {
+    type: Number,
+    optional: true,
+    viewableBy: ['guests'],
+    canUpdate: ['admins', 'sunshineRegiment'],
+    canCreate: ['admins', 'sunshineRegiment'], 
+  }
+};
+
+export const TagFlags = createCollection({
+  collectionName: 'TagFlags',
+  typeName: 'TagFlag',
+  schema,
+  resolvers: getDefaultResolvers('TagFlags'),
+  mutations: getDefaultMutations('TagFlags'),
```

Comment:
Should this be restricted so only sunshines/admins can create tag flags?

Reply:
Yeah, seems right. All the fields are restricted, so this would just allow someone to do create a dysfunctional tagFlag